contracts/Depository.sol

```diff
@@ -100,21 +135,39 @@ contract Depository is IErrorsTokenomics {
     mapping(uint256 => Product) public mapBondProducts;
 
     /// @dev Depository constructor.
+    /// @param _name Service contract name.
+    /// @param _symbol Agent contract symbol.
+    /// @param _baseURI Agent registry token base URI.
     /// @param _olas OLAS token address.
     /// @param _treasury Treasury address.
     /// @param _tokenomics Tokenomics address.
-    constructor(address _olas, address _tokenomics, address _treasury, address _bondCalculator)
+    constructor(
+        string memory _name,
+        string memory _symbol,
+        string memory _baseURI,
+        address _olas,
+        address _tokenomics,
+        address _treasury,
+        address _bondCalculator
+    )
+        ERC721(_name, _symbol)
     {
-        owner = msg.sender;
-
         // Check for at least one zero contract address
         if (_olas == address(0) || _tokenomics == address(0) || _treasury == address(0) || _bondCalculator == address(0)) {
             revert ZeroAddress();
         }
+
+        // Check for base URI zero value
+        if (bytes(_baseURI).length == 0) {
+            revert ZeroValue();
+        }
+
         olas = _olas;
         tokenomics = _tokenomics;
         treasury = _treasury;
         bondCalculator = _bondCalculator;
+        baseURI = _baseURI;
```

Comment:
Doesn't make sense here imo. We can have a purely on-chain NFT. There are these SVG libraries that are pretty cool. We don't want to have to serve custom images here, just unnecessary.

Reply:
Actually we don't need any link at all. All we need to comply with the standard is that the function `tokenURI()` does not revert. So it can be as simple as just returning an NFT id as a string.